doc/user/content/sql/create-source/_index.md

```diff
@@ -68,24 +68,31 @@ Materialize supports all [Avro types](https://avro.apache.org/docs/current/spec.
 Support for the more ergonomic `FORMAT JSON` is in progress {{% gh 7186 %}}!
 {{</ note >}}
 
-Materialize cannot decode JSON directly from an external data source. Instead, you must
-create a source that reads the data as [raw bytes](#bytes), and then handle the conversion to [`jsonb`](/sql/types/jsonb) using an intermediate view:
+Materialize cannot decode JSON directly from an external data source. Instead, you must create a source that reads the data as [raw bytes](#bytes), and then handle the conversion to [`jsonb`](/sql/types/jsonb) using view. We also recommend parsing that `jsonb` into primitive types using a materialized view so that subsequent queries optimize performance and avoid redundant parsing.
 
 ```sql
-CREATE SOURCE bytea_source
+-- create raw byte array source
+CREATE SOURCE my_bytea_source
   FROM ...
   FORMAT BYTES
   WITH (SIZE='3xsmall');
 
-CREATE VIEW jsonified_source AS
+-- convert from byte array to text to jsonb
+CREATE VIEW my_jsonb_source AS
   SELECT
-    data->>'field1' AS field_1,
-    data->>'field2' AS field_2,
-    data->>'field3' AS field_3
-  FROM (SELECT CONVERT_FROM(data, 'utf8')::jsonb AS data FROM bytea_source);
+    CONVERT_FROM(data, 'utf8')::jsonb AS data
+  FROM bytea_source;
+
```

Comment:
```suggestion

To avoid redundant processing and ensure a typed representation of the source is available across clusters, you should create a [materialized view](/overview/key-concepts/#materialized-views).
```